ARCHITECTURE.md

```diff
@@ -1,6 +1,266 @@
+  - **time**: The specific time of day the trigger should activate.
+
+
+#### Action
+
+An action object defines the operations that are performed when the workflow is triggered.
+
+- **type**: (String) The type of action. This should correspond to a predefined action type in the application.
+- **options**?: (Object) An optional set of parameters specific to the action type.
+
+```json
+{
+  "type": "Application.MultiSender",
+  "options": {
+    "items": [
+      {
+        "asset": {
+          "address": "0xTokenAddress",
+          "decimals": 18
+        },
+        "amount": "100",
+        "to": "0xRecipientAddress"
+      },
+      {
+        "asset": null,
+        "amount": "0.1",
+        "to": "0xAnotherRecipientAddress"
+      }
+    ]
+  }
+}
+```
+
+- **type**: Specifies a multi-send action by the application.
+- **options**:
+  - **items**: An array of objects representing the assets to be sent, each containing:
+    - **asset**?: An optional object representing the ERC20 token.
+      - **address**: The address of the ERC20 token.
+      - **decimals**: The number of decimals for the ERC20 token.
+    - **amount**: The amount of the asset to send.
+    - **to**: The recipient address.
+
+
+### Extending Configuration
+
+This format is designed to be extensible. Additional triggers and actions can be defined by specifying new types and corresponding options. This approach allows for a highly customizable and adaptable workflow configuration system.
+
+```ts
+const wf = workflowFactory
+  // Registers an action with the workflow using the action class directly
+  .registerAction(MultiSenderAction) // => new MultiSenderAction(options)
+  // Registers a trigger with the workflow using the trigger class directly
+  .registerTrigger(PriceBasedTrigger) // => new PriceBasedTrigger(options, commonConfig)
+```
+
+
+### Using Configuration
+
+Below is a complete example of a workflow configuration:
+
+```ts
+const workflowFactory = new WorkflowsFactory(provider);
+
+const config = {
+  name: 'Send assets to multiple addresses',
+  issuer: 'Ditto',
+  hint: 'Send your assets to multiple addresses in one transaction',
+  triggers: [
+    {
+      type: 'Core.Scheduled',
+      options: {
+        frequency: 'daily',
+        time: '08:00'
+      }
+    }
+  ],
+  actions: [
+    {
+      type: 'Application.MultiSender',
+      options: {
+        items: [
+          {
+            asset: {
+              address: '0xTokenAddress',
+              decimals: 18
+            },
+            amount: '100',
+            to: '0xRecipientAddress'
+          },
+          {
+            asset: null,
+            amount: '0.1',
+            to: '0xAnotherRecipientAddress'
+          }
+        ]
+      }
+    }
+  ],
+};
+
+const wf = workflowFactory
+  .registerAction(MultiSenderAction)
+  .registerTrigger(ScheduledTrigger)
+  // Creates a workflow from the given configuration object.
+  .createFromConfig(config);
```

Comment:
I didn't understand why this config is needed. We already have registered actions & triggers, why do we need another config?